tox.ini

```
@@ -0,0 +1,14 @@
+[tox]
+requires =
+    tox>=4.2
+env_list =
+    py{312, 311, 310, 39, 38}
+
+[testenv]
+pass_env =
+    FORCE_COLOR
+commands_pre =
+    {envpython} -m pip install -U -r dev-requirements.txt
+commands =
+    {envpython} -m pytest \
+      {posargs}
```

Nit: do we need to have this split into two lines?

```suggestion
    {envpython} -m pytest {posargs}
```